### PR TITLE
Add PriorityClassName to the CNPG Controller

### PR DESCRIPTION
**Pull request by jakubhajek**
Fix: #64 

This PR adds the capability to define a `priorityClassName` in the CNPG deployment. Since the controller is mission critical workload by defining a priorityClassName we can ensure that controller gets enough resources defined in the `priortiyClass`.

### CODE REVIEW

File: charts/cloudnative-pg/Chart.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ name: cloudnative-pg
 description: CloudNativePG Helm Chart
 icon: https://raw.githubusercontent.com/cloudnative-pg/artwork/main/cloudnativepg-logo.svg
 type: application
-version: "0.16.0"
+version: "0.16.1"
```
Comment (phisco) [inline charts/cloudnative-pg/Chart.yaml:21]: The release procedure will take care of bumping the version:
```suggestion
version: "0.16.0"
```

File: charts/cloudnative-pg/templates/deployment.yaml [hunk 2]
```diff
@@ -87,6 +87,9 @@ spec:
             scheme: HTTPS
         resources:
           {{- toYaml .Values.resources | nindent 10 }}
+        {{- if .Values.priorityClassName }}
+        priorityClassName: {{ .Values.priorityClassName }}
+        {{- end }}
```
Comment (phisco) [inline charts/cloudnative-pg/templates/deployment.yaml:92]: priorityClassName should be at the root of the podSpec
```suggestion
```

**Comment by jakubhajek** [comment]
Hello @phisco, The PR has been updated including your suggestions. Thank you for your review!

**Comment by phisco** [comment]
Hi @jakubhajek, you forgot to remove the container level priorityClassName. Then, can you squash all commits to a single commit compliant with conventional commits? Something like "feat: customizable operator's priorityClassName" will work. Thank you for contributing!

**Review by phisco — APPROVED** [review]

**Review by sxd — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

**Review by leonardoce — APPROVED** [review]

**Pull request merged**